### PR TITLE
Implement range_de for SnapshotMap

### PR DESCRIPTION
**Pull request by uint**
Deals with #461, but doesn't close it

### CODE REVIEW

File: packages/storage-plus/src/snapshot/map.rs [hunk 1]
```diff
@@ -373,4 +431,70 @@ mod tests {
...
+        let mut store = MockStorage::new();
+        init_data(&EVERY, &mut store);
+
+        // let's try to iterate!
+        let all: StdResult<Vec<_>> = EVERY
+            .range_de(&store, None, None, Order::Ascending)
+            .collect();
+        let all = all.unwrap();
+        assert_eq!(2, all.len());
+        assert_eq!(all, vec![(b"C".to_vec(), 13), (b"D".to_vec(), 22)]);
```
Comment (ethanfrey) [inline packages/storage-plus/src/snapshot/map.rs:449]: Nice demo showing the deserialisation.

**Review by ethanfrey — COMMENTED** [review]
Took a quick look and it seems correct to me.

Maybe @maurolacy can give some feedback as you work on porting this to the IndexedMap types. (And I guess we would want such range_de on the indexes??? or maybe that is another PR?)

File: packages/storage-plus/src/snapshot/map.rs [hunk 2]
```diff
@@ -373,4 +431,70 @@ mod tests {
...
+
+        let mut store = MockStorage::new();
+        init_data(&EVERY, &mut store);
+
+        // let's try to iterate!
+        let all: StdResult<Vec<_>> = EVERY
+            .range_de(&store, None, None, Order::Ascending)
+            .collect();
+        let all = all.unwrap();
+        assert_eq!(2, all.len());
```
[the remark this answers is not in the document]
Comment (uint) [reply packages/storage-plus/src/snapshot/map.rs:448]: Fair. We do this in other places I think.

[the remark this answers is not in the document]
Comment (ethanfrey) [reply packages/storage-plus/src/snapshot/map.rs:448]: Yes, we started removing some of those in recent PR reviews, which is why I point this out.

Nice to read other PR reviews also to see discussions on APIs and best practices... I think that is where the code style really gets hashed out

**Pull request merged**